MDS_3/include/CGAL/tetrahedron_soup_to_triangulation_3.h

```diff
@@ -0,0 +1,166 @@
+    std::map<std::array<int, 3>, typename Tr::Cell::Surface_patch_index> border_facets;
+    std::map<Point, int> p2i;
+
+    CGAL_assertion_code(
+      typename Triangulation::Geom_traits::Orientation_3 orientation =
+      tr.geom_traits().orientation_3_object();
+    );
+
+    for (typename TetrahedronRange::value_type tet : tets)
+    {
+      CGAL_assertion(tet.orientation() == CGAL::POSITIVE);
+      std::array<int, 5> cell;
+
+      for (int i = 0; i < 4; ++i)
+      {
+        const Point& pi = tet[i];
+        if (p2i.find(pi) == p2i.end())
+        {
+          points.push_back(pi);
+          int index = static_cast<int>(points.size() - 1);
+          p2i.insert(std::make_pair(pi, index));
+          cell[i] = index;
+        }
+        else
+          cell[i] = p2i.at(pi);
+      }
+      cell[4] = 1;
+
+      CGAL_assertion(orientation(points[cell[0]],
+        points[cell[1]], points[cell[2]], points[cell[3]]) == CGAL::POSITIVE);
+
+      finite_cells.push_back(cell);
+    }
+
+    CGAL::MDS_3::build_triangulation(tr, points, finite_cells, border_facets);
+
+    CGAL_assertion(CGAL::MDS_3::internal::is_convex(tr));
+  }
+
+  /** \ingroup PkgMDS3Functions
+  * builds a 3D triangulation from a soup of tetrahedra.
+  *
+  * @tparam PointRange  a model of the concept `RandomAccessContainer`
+  * whose value type is the point type
+  * @tparam TetrahedronRange a model of the concept `RandomAccessContainer` whose
+  * value type is a model of the concept `RandomAccessContainer` whose value type is `std::size_t`
+  * @tparam Triangulation a valid triangulation class that has
+  * a vertex base model of `MeshVertexBase_3`
+  * and a cell base model of `MeshCellBase_3`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param points points of the soup of tetrahedra
+  * @param tets each element in the range describes a tetrahedron using the indices of the points
+  * in `points` (indices 0 to 3), and the associated `Subdomain_index` (index 4)
+  * @param tr the 3D triangulation to be built
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{surface_facets}
+  *     \cgalParamDescription{each element in the range describes a surface facet using the indices of points
+  *       in `points` (indices 0 to 2), and the associated `Surface_patch_index` (index 3)}
+  *     \cgalParamType{a class model of `AssociativeContainer`
+  *                    whose key type is model of `RandomAccessContainer`
+  *                    and mapped type is `Tr::Cell::Surface_patch_index`}
+  *     \cgalParamDefault{An empty `std::map<std::array<int, 3>, typename Tr::Cell::Surface_patch_index>`}
+  *     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+  *                     must be available in `PolygonMesh`.}
```

Comment:
What is the relationship between `np::surface_facets` and "an internal property map for `CGAL::vertex_point_t`"?